### PR TITLE
refactor(service): moved edit and unlink endpoints to controllers

### PR DESCRIPTION
**Pull request by jsam**
This PR fixes syncing problems in edit and unlink endpoints and also moves them to the new structure using controllers. In addition, it applies small fixes to activity schema and cwl migrations.

addresses: #1566

### CODE REVIEW

File: tests/service/views/test_dataset_views.py [hunk 1]
```diff
@@ -902,7 +902,7 @@ def test_add_remote_and_local_file(svc_client_with_repo):
 
 @pytest.mark.service
 @pytest.mark.integration
-@flaky(max_runs=10, min_passes=1)
+@flaky(max_runs=1, min_passes=1)
```
Comment (Panaetius) [inline tests/service/views/test_dataset_views.py:905]: left-over from debugging?

**Review by Panaetius — CHANGES_REQUESTED** [review]
Looks great, just one minor comment

File: tests/service/views/test_dataset_views.py (hunk 1 above)
Comment (jsam) [reply tests/service/views/test_dataset_views.py]: Thanks. Fixed.

**Review by Panaetius — APPROVED** [review]
🎉

**Pull request merged**